File: frontend/src/api/model/rest-error.ts
import Dictionary from '../../utils/dictionary';

export interface RestError {
    status: number;
    code: string;
    issues: RestIssue[];
}

export interface RestIssue {
    code: string;
    details: string;
    field?: string;
    timestamp: Date;
    meta: Dictionary<any>;
}

File: backend/src/main/kotlin/com/idleon/glacierfrostbuilds/api/exceptions/RestError.kt
package com.idleon.glacierfrostbuilds.api.exceptions

import org.springframework.http.HttpStatus
import java.time.LocalDateTime


class RestException(val error: RestError) : Exception()

data class RestError(
    val status: HttpStatus,
    val code: String,
    val errors: List<RestIssue> = arrayListOf()
)

data class RestIssue(
    val code: String,
    val details: String,
    val timestamp: LocalDateTime,
)
